setup.cfg

```diff
@@ -1,2 +1,5 @@
 [tool:pytest]
 testpaths = tests
+filterwarnings =
+    ignore::rasterio.errors.NotGeoreferencedWarning
+    ignore::rasterio.errors.RasterioDeprecationWarning
```

Comment:
@geowurster ignoring custom categories was my first thought, but I ran into an obscure issue: the warnings filter is set up before any imports and so you can only filter on builtin warning categories. For custom warnings, we have to filter based on the message.

Reply:
@sgillies Did you test against the most recent version of `pytest` (`3.2.3`)?  It works for me, although looking through the [changelog](https://docs.pytest.org/en/latest/changelog.html) I don't see any recent changes that would explain the behavior change.

Here's what I get on this branch:

```console
pytest tests/test_blocks.py
Test data present.
=========================================== test session starts ===========================================
platform darwin -- Python 3.6.3, pytest-3.2.3, py-1.4.34, pluggy-0.4.0
rootdir: /Users/wursterk/code/rasterio, inifile: setup.cfg
plugins: cov-2.5.1, hypothesis-3.32.1
collected 17 items                                                                                         

tests/test_blocks.py .................

======================================== 17 passed in 0.10 seconds ========================================
```

versus `master`:

```console
$ pytest tests/test_blocks.py
Test data present.
=========================================== test session starts ===========================================
platform darwin -- Python 3.6.3, pytest-3.2.3, py-1.4.34, pluggy-0.4.0
rootdir: /Users/wursterk/code/rasterio, inifile: pytest.ini
plugins: cov-2.5.1, hypothesis-3.32.1
collected 17 items                                                                                         

tests/test_blocks.py .................

============================================ warnings summary =============================================
tests/test_blocks.py::WindowTest::test_eval
  /Users/wursterk/code/rasterio/rasterio/__init__.py:40: FutureWarning: Deprecated; Use rasterio.windows instead
    warnings.warn("Deprecated; Use rasterio.windows instead", FutureWarning)

tests/test_blocks.py::WindowTest::test_window_shape_None_start
  /Users/wursterk/code/rasterio/rasterio/__init__.py:46: FutureWarning: Deprecated; Use rasterio.windows instead
    warnings.warn("Deprecated; Use rasterio.windows instead", FutureWarning)

tests/test_blocks.py::WindowTest::test_window_shape_None_stop
  /Users/wursterk/code/rasterio/rasterio/__init__.py:46: FutureWarning: Deprecated; Use rasterio.windows instead
    warnings.warn("Deprecated; Use rasterio.windows instead", FutureWarning)

tests/test_blocks.py::WindowTest::test_window_shape_negative
  /Users/wursterk/code/rasterio/rasterio/__init__.py:46: FutureWarning: Deprecated; Use rasterio.windows instead
    warnings.warn("Deprecated; Use rasterio.windows instead", FutureWarning)

tests/test_blocks.py::WindowTest::test_window_shape_positive
  /Users/wursterk/code/rasterio/rasterio/__init__.py:46: FutureWarning: Deprecated; Use rasterio.windows instead
    warnings.warn("Deprecated; Use rasterio.windows instead", FutureWarning)

tests/test_blocks.py::test_window_index
  /Users/wursterk/code/rasterio/rasterio/__init__.py:52: FutureWarning: Deprecated; Use rasterio.windows instead
    warnings.warn("Deprecated; Use rasterio.windows instead", FutureWarning)

tests/test_blocks.py::WindowReadTest::test_read_window
  /Users/wursterk/code/rasterio/rasterio/__init__.py:46: FutureWarning: Deprecated; Use rasterio.windows instead
    warnings.warn("Deprecated; Use rasterio.windows instead", FutureWarning)

-- Docs: http://doc.pytest.org/en/latest/warnings.html
================================== 17 passed, 7 warnings in 0.11 seconds ==================================
```

Reply:
Maybe there's a difference between `python -W ...` where I hit the problem and `python -m pytest -W ...`? Since it's working for you, let's give it a shot.

Reply:
Just checked, this nicely quieted messages for me.